src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2688,14 +2599,17 @@ private void VisitStatementsWithLocalFunctions(BoundBlock block)
             // variables set according to Joining the state at all the
             // local function use sites
             var state = TopState();
-            for (int slot = 1; slot < localFunctionState.StartingState.Capacity; slot++)
-            {
-                var symbol = variableBySlot[RootSlot(slot)].Symbol;
-                if (Symbol.IsCaptured(symbol, localFunc))
+            var startingState = localFunctionState.StartingState;
+            startingState.ForEach(
+                (slot, variables) =>
```

Comment:
Consider marking this lambda static, or making it a regular foreach loop if it can't be.

Reply:
Unfortunately, there are several locals from the containing method, `state`, `startingState`, and `localFunc`, and `state` would need to be passed by reference, so it seemed cleaner to allow a closure class instance to be created, especially since this method is only called for local function declarations.

---
In reply to: [560551991](https://github.com/dotnet/roslyn/pull/50417#discussion_r560551991) [](ancestors = 560551991)

Reply:
Then why not just make this a regular `foreach` loop?

---
In reply to: [560600004](https://github.com/dotnet/roslyn/pull/50417#discussion_r560600004) [](ancestors = 560600004,560551991)

Reply:
> Then why not just make this a regular `foreach` loop? 

The method traverses `LocalState` at all scopes, so a simple `foreach` is not sufficient.

---
In reply to: [561181556](https://github.com/dotnet/roslyn/pull/50417#discussion_r561181556) [](ancestors = 561181556,560600004,560551991)